apis/topology/v1beta1/topology_types.go

```diff
@@ -211,3 +211,77 @@ func EnsureDeletedTopologyRef(
 	}
 	return ctrl.Result{}, nil
 }
+
+// Given a StatefulSet/Deployment/DaemonSet PodTemplateSpec, apply the referenced
+// topology
+func (t Topology) ApplyTo(
+	pod *corev1.PodTemplateSpec,
+) {
+	// Get the Topology .Spec
+	ts := t.Spec
+	// Process TopologySpreadConstraints if defined in the referenced Topology
+	if ts.TopologySpreadConstraints != nil {
+		pod.Spec.TopologySpreadConstraints = *t.Spec.TopologySpreadConstraints
+	}
+	// Process Affinity if defined in the referenced Topology
+	if ts.Affinity != nil {
+		pod.Spec.Affinity = ts.Affinity
+	}
```

Comment:
@olliewalsh do you think we should include here the `DistributePods` part? I didn't add it because of DaemonSets that would use the same function, but let me know if you have thoughts around this

Reply:
No, I don't think we should include it, at least not in this function